src/main/java/com/minecolonies/coremod/commands/CommandEntryPointNew.java

```diff
@@ -0,0 +1,596 @@
+            {
+                final TreeNode<Menu> childTreeNode = buildMenu(childMenuType, menuTypesSoFar);
+                treeNode.addChild(childTreeNode);
+            }
+        }
+
+        return treeNode;
+    }
+
+    @NotNull
+    @Override
+    public String getName()
+    {
+        return root.getData().getMenuItemName();
+    }
+
+    @NotNull
+    @Override
+    public String getUsage(final ICommandSender sender)
+    {
+        return getCommandUsage(sender, root);
+    }
+
+    @NotNull
+    public String getCommandUsage(final ICommandSender sender, final TreeNode<Menu> currentMenuTreeNode)
+    {
+        final MenuType currentMenuType = currentMenuTreeNode.getData().getMenuType();
+        if (currentMenuTreeNode.hasChildren())
+        {
+            final StringBuilder sb = new StringBuilder();
+            buildParentPathString(sb, currentMenuTreeNode);
+            sb.append(" <");
+            boolean first = true;
+            for (final TreeNode<Menu> childTreeNode : currentMenuTreeNode.getChildren())
+            {
+                if (first)
+                {
+                    first = false;
+                }
+                else
+                {
+                    sb.append('|');
+                }
+                sb.append(childTreeNode.getData().getMenuItemName());
+            }
+            sb.append('>');
+            return sb.toString();
+        }
+        else
+        {
+            final StringBuilder sb = new StringBuilder();
+            buildParentPathString(sb, currentMenuTreeNode);
+            final Menu menu = currentMenuType.getMenu();
+            if (!currentMenuType.isNavigationMenu())
+            {
+                final ActionMenu actionMenu = (ActionMenu) menu;
+                for (final ActionArgument actionArgument : actionMenu.getActionArgumentList())
+                {
+                    sb.append(' ').append(actionArgument.getUsage());
+                }
+            }
+            return sb.toString();
+        }
+    }
+
+    private void buildParentPathString(final StringBuilder sb, final TreeNode<Menu> menuTreeNode)
+    {
+        final List<TreeNode<Menu>> treeNodeMenuListMinusRoot = new ArrayList<>();
+        TreeNode<Menu> currentMenuTreeNode = menuTreeNode;
+        while (null != currentMenuTreeNode.getParent())
+        {
+            treeNodeMenuListMinusRoot.add(currentMenuTreeNode);
+            currentMenuTreeNode = currentMenuTreeNode.getParent();
+        }
+        sb.append('/').append(currentMenuTreeNode.getData().getMenuItemName());
+
+        Collections.reverse(treeNodeMenuListMinusRoot);
+        for (final TreeNode<Menu> treeNode : treeNodeMenuListMinusRoot)
+        {
+            sb.append(' ').append(treeNode.getData().getMenuItemName());
+        }
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String[] args) throws CommandException
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This method has 105 lines, which is greater than the 100 lines authorized. Split it into smaller methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS138)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "execute" is 21 which is greater than 15 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 63 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)